### PR TITLE
Added a basic guide for use with TypeScript

### PR DESCRIPTION
**Pull request by joshtynjala**

### CODE REVIEW

**Comment by atrauzzi** [comment]
Is there any concern over the requirement that the component managed by Reach has to take a coupling to it?

What if I'm adding a component that isn't part of my project?  I can't force it to use the reach props interface.

**Comment by joshtynjala** [comment]
Is this the best place to ask that question? Seems like someone not using TypeScript could easily have the same concern.

**Comment by atrauzzi** [comment]
Not necessarily. In JS, the component will happily work, totally oblivious to how it's being run.

Yes there may be design implications, if intended. But we haven't ascertained that yet. And if there are, then yeah, this is a great place to get those intentions noted explicitly.

**Comment by joshtynjala** [comment]
It's possible that the type definitions have been changed by the community since I last installed them, but I don't think I was required to use the RouteComponentProps interface. I could opt in if I needed the props, but any component was allowed. Same as JS.

**Comment by atrauzzi** [comment]
If you don't include the interface in your component's props, you run into https://github.com/reach/router/issues/141 .

This is not a problem in plain JavaScript obviously _because duck-punching_.  In TypeScript however, it's wanting you to declare the presence of a new prop.

Because of this difference, in the JS world, you can use reach with any component, without anything trying to suggest that it's a possible error.  In TS, it's explicitly an error.

The biggest drawback I see is that it'll force people to write arbitrary wrappers for code they don't control, just to satisfy the compiler.  This would only happen because they're using TypeScript, which might at that point seems to me to make it wrong.

The point overall here is that I think this still remains an unresolved question for reach.  Any documentation should at least be pointing out the presence of this difference.  And ideally any solution or conclusions (as they happen).

**Comment by joshtynjala** [comment]
> If you don't include the interface in your component's props, you run into #141 .

Oh yes, I forgot about that. I haven't had to touch the routing related parts of my app's code in a while.

File: website/src/markdown/pages/typescript.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
+# TypeScript
+
+Reach Router may be used with [TypeScript](https://www.typescriptlang.org/).
+
+## Install Typings
+
+```sh
+npm install @types/reach__router --save-dev
+# or
+yarn add @reach/reach__router --dev
```
[the remark this answers is not in the document]
Comment (joshtynjala) [reply website/src/markdown/pages/typescript.md]: Fixed. Thanks!

**Pull request merged**